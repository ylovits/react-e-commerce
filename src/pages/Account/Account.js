import React, { Fragment } from "react";
import { Link } from "react-router-dom";
import { auth } from "../../firebase/firebase.utils";
import { connect } from "react-redux";

const Account = ({ currentUser }) => {
  return (
    <Fragment>
      <div className="bg-dark text-light mb-4 p-2">
        Welcome <strong>{currentUser && currentUser.displayName}</strong>
      </div>
      <div className="row justify-content-center">
        <div className="bg-light account mb-4 p-2">
          <h3 className="mb-3">Manage your account</h3>

          {currentUser ? (
            <button
              className="btn btn-lg btn-primary btn-block text-uppercase mb-2"
              onClick={() => auth.signOut()}
            >
              Sign Out
            </button>
          ) : (
            <Link
              className="btn btn-lg btn-primary btn-block text-uppercase mb-2"
              to="/signin"
            >
              Sign In
            </Link>
          )}
          {currentUser && (
            <button className="btn btn-lg btn-secondary btn-block text-uppercase mb-2">
              Delete Account
            </button>
          )}
        </div>
      </div>
    </Fragment>
  );
};

const mapStateToProps = state => {
  return {
    currentUser: state.user.currentUser
  };
};

export default connect(mapStateToProps)(Account);
