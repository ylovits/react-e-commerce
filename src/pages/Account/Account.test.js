import React from "react";
import { shallow } from "enzyme";
import { Provider } from "react-redux";
import Account from "./Account";

import configureMockStore from "redux-mock-store";
const mockStore = configureMockStore();
const store = mockStore({});

it("<Account /> rendered", () => {
  expect(
    shallow(
      <Provider store={store}>
        <Account />
      </Provider>
    )
  ).toMatchSnapshot();
});
