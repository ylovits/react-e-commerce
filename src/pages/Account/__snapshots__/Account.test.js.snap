// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Account /> rendered 1`] = `ShallowWrapper {}`;
